### ISSUE TITLE
zfs list -p -o compressratio outputs with 'x', but it shouldn't

### ISSUE THREAD

**Issue by serg-ku** [zfsonlinux/zfs#7901]
### System information

Type | Version/Name
 --- | --- 
Distribution Name	| Centos
Distribution Version	| 7.5
Linux Kernel	| 3.10.0-862.11.6.el7.x86_64
Architecture	| x86_64
ZFS Version	| v0.7.10
SPL Version	| v0.7.10

### Describe the problem you're observing
zfs list output in parsable(-p) form displaying compressratio with 'x' at the end, same as without '-p' , but I think it should behave like zpool list -p dedupratio

### Describe how to reproduce the problem
```
# zfs list -o compressratio
RATIO
1.00x
# zfs list -o compressratio -p
RATIO
1.00x
```
in zpool list:
```
# zpool list -o dedupratio
DEDUP
1.00x
# zpool list -o dedupratio -p
DEDUP
 1.00
```

**Comment by loli10K** [comment]
Actually this is already fixed in master (https://github.com/zfsonlinux/zfs/commit/b0bd8ffecd70b46e584e2e52ea74f2373b544217): we could cherry-pick the fix into zfs-0.7-release, if necessary.

**Comment by behlendorf** [comment]
Closing.  Already fixed, we can consider this as a possible thing to cherry-pick for 0.7.
